adyenv6core/src/com/adyen/v6/exceptions/AdyenNonAuthorizedPaymentException.java

```diff
@@ -35,6 +37,10 @@ public AdyenNonAuthorizedPaymentException(PaymentsResponse paymentsResponse) {
         this.paymentsResponse = paymentsResponse;
     }
 
+    public AdyenNonAuthorizedPaymentException(TerminalAPIResponse terminalApiResponse) {
```

Comment:
Do we need this ?

Reply:
It's needed to handle the FAILURE cases, since we will need some information from the response object (errorConditions, etc)